### PR TITLE
test(sbb-form-field): add visual spec

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
To discuss: reduce test cases?

### CODE REVIEW

File: src/elements/form-field/form-field/form-field.visual.spec.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        disabled: state.disabled,
+        readonly: state.readonly,
+      };
+
+      for (const [name, template] of component.entries()) {
+        it(
+          `slot=none ${name} ${visualDiffDefault.name}`,
+          visualDiffDefault.with(async (setup) => {
+            await setup.withFixture(html`${formField(args, template(args))}`, {
+              backgroundColor: negative ? '#484040' : undefined,
```
Comment (jeripeierSBB) [inline src/elements/form-field/form-field/form-field.visual.spec.ts:198]:
```suggestion
              backgroundColor: negative ? 'var(--sbb-color-black)' : undefined,
```

File: src/elements/form-field/form-field/form-field.visual.spec.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+            });
+          }),
+        );
+
+        it(
+          `slot=buttons ${name} ${visualDiffDefault.name}`,
+          visualDiffDefault.with(async (setup) => {
+            const templateResult: TemplateResult = html`${template(args)} ${buttonsAndPopover(args)}`;
+            await setup.withFixture(html`${formField(args, templateResult)}`, {
+              backgroundColor: negative ? '#484040' : undefined,
```
Comment (jeripeierSBB) [inline src/elements/form-field/form-field/form-field.visual.spec.ts:208]:
```suggestion
              backgroundColor: negative ? 'var(--sbb-color-black)' : undefined,
```

File: src/elements/form-field/form-field/form-field.visual.spec.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
+import { html, nothing, type TemplateResult } from 'lit';
```
Comment (jeripeierSBB) [inline src/elements/form-field/form-field/form-field.visual.spec.ts:1]: I'm not sure whether the focus is always where it's meant to be. Do you really want to test the mini button focus or was the goal to focus the input. In the second case you could use the method "withStateElement()" and pass the input element. there would be an example in the typography.visual.spec.ts file.

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Nice work! Basically I think it's ok if there are a lot of combinations. It's in the nature of the form field that there are a lot of possible combinations and visual outcomes.

File: src/elements/form-field/form-field/form-field.visual.spec.ts (hunk 3 above)
Comment (DavideMininni-Fincons) [reply src/elements/form-field/form-field/form-field.visual.spec.ts:2]: Removed the focus from the states array and moved the tests with button/popover apart

**Review by jeripeierSBB — APPROVED** [review]
LGTM, not sure whether the active and focus state should also be tested with negative? I tend to a yes
